### PR TITLE
add Parity trace functions

### PR DESCRIPTION
**Pull request by jennyailin**
add functions from https://github.com/paritytech/parity/wiki/JSONRPC-trace-module

### CODE REVIEW

File: web3/trace.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
+from web3 import formatters
+from web3.utils.empty import (
+    empty,
+)
+
+class Trace(object):
+    def __init__(self, web3):
+        self.web3 = web3
+
```
Comment (carver) [inline web3/trace.py]: Using `class Trace(Module):` means that the `__init__` method can be deleted. Yay for less code! (requires importing `from web3.module import Module`)

File: web3/trace.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    defaultAccount = empty
```
Comment (carver) [inline web3/trace.py]: Looks like this isn't used and can be deleted? (which means the import of `empty` can be, too)

File: web3/trace.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
+from web3 import formatters
```
Comment (carver) [inline web3/trace.py]: This module is gone.

File: web3/trace.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+class Trace(object):
+    def __init__(self, web3):
+        self.web3 = web3
+
+    defaultAccount = empty
+    defaultBlock = "latest"
+
+    def call(self, transaction, types, block_identifier=None):
+        formatted_transaction = formatters.input_transaction_formatter(self, transaction)
```
Comment (carver) [inline web3/trace.py]: This function has been refactored away. It looks like adding a two new request formatters for `'trace_call'` to `web3/middleware/pythonic.py` is the right path forward here.

File: web3/trace.py [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        if block_identifier is None:
+            block_identifier = self.defaultBlock
+        return self.web3._requestManager.request_blocking(
```
Comment (carver) [inline web3/trace.py]: All of the `_requestManager` properties are now just `manager`

**Comment by allquantor** [comment]
Hey @jennyailin do you still think to merge this? Is is an valuable addition IMO.

**Comment by jennyailin** [comment]
@allquantor I do plan on it, but unsure when I can get to it. Hopefully in a few weeks :)

**Comment by carver** [comment]
Looks like #850 took care of most/all of this. Feel free to comment or reopen if any functionality is still missing.  Several of the patterns from this PR made their way into #850. Thank you for the contribution!

**Pull request closed without merging**